src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,179 @@
+const ActiveDescendant = require('makeup-active-descendant');
+const Expander = require('makeup-expander');
+const elementScroll = require('../../common/element-scroll');
+const emitAndFire = require('../../common/emit-and-fire');
+const eventUtils = require('../../common/event-utils');
+const observer = require('../../common/property-observer');
+const safeRegex = require('../../common/build-safe-regex');
+
+module.exports = require('marko-widgets').defineComponent({
+    template: require('./template.marko'),
+    getInitialProps(input) {
+        return Object.assign({
+            options: []
+        }, input);
+    },
+    getInitialState(input) {
+        const index = findIndex(input.options, option => option.selected);
+        const currentValue = input.options[index] && input.options[index].text;
+
+        return Object.assign({}, input, {
+            selectedIndex: index === -1 ? null : index,
+            currentValue: currentValue
+        });
+    },
+    init() {
+        observer.observeRoot(this, ['disabled'], () => {
+            this.expander.expandOnClick = !this.state.disabled;
+        });
+
+        this.getEls('option').forEach((optionEl, i) => {
+            Object.defineProperty(optionEl, 'selected', {
+                get: () => this.state.selectedIndex === i,
+                set: (value) => this.setSelectedIndex(value ? i : null)
+            });
+        });
+    },
+    onRender() {
+        const comboboxInput = this.getEl('input');
+        const optionsContainer = this.getEl('options');
+        const selectedIndex = findIndex(this.state.options, option => option.selected);
+
+        if (this.state.options.length) {
+            this.activeDescendant = ActiveDescendant.createLinear(
+                this.el,
+                comboboxInput,
+                optionsContainer,
+                '.combobox__option[role=option]', {
+                    activeDescendantClassName: 'combobox__option--active',
+                    autoInit: selectedIndex === -1 ? -1 : 0,
+                    autoReset: -1
+                }
+            );
+
+            this.expander = new Expander(this.el, {
+                autoCollapse: true,
+                expandOnFocus: true,
+                expandOnClick: this.state.readonly && !this.state.disabled,
+                collapseOnFocusOut: !this.state.readonly,
+                contentSelector: `#${optionsContainer.id}`,
+                hostSelector: `#${this.getEl('input').id}`,
+                expandedClass: 'combobox--expanded',
+                simulateSpacebarClick: true
+            });
+        }
+
+        this.moveCursorToEnd();
+    },
+    onBeforeUpdate() {
+        if (this.activeDescendant) {
+            this.activeDescendant.destroy();
+        }
+
+        if (this.expander) {
+            this.expander.cancelAsync();
+        }
+    },
+    onDestroy() {
+        this.activeDescendant.destroy();
```

Comment:
We may need to add the same guards for `this.activeDescendant` and `this.expander` in the `onDestroy` as well since you could render an empty combobox, and then remove it which would cause an error.

I had a similar issue in the dialog and opted to create a `_release` method (https://github.com/eBay/ebayui-core/blob/8e2d0e4987e1b7245097227429b2bed301fac36f/src/components/ebay-dialog/index.js#L134) which was then called in both places: https://github.com/eBay/ebayui-core/blob/8e2d0e4987e1b7245097227429b2bed301fac36f/src/components/ebay-dialog/index.js#L28 & https://github.com/eBay/ebayui-core/blob/8e2d0e4987e1b7245097227429b2bed301fac36f/src/components/ebay-dialog/index.js#L33

Might make sense to do something similar here.